integration/kubernetes/init.sh

```diff
@@ -67,127 +100,208 @@ build_custom_stress_image()
 	fi
 }
 
-cri_runtime="${CRI_RUNTIME:-crio}"
-kubernetes_version=$(get_version "externals.kubernetes.version")
+# Delete the CNI configuration files and delete the interface.
+# That's needed because `kubeadm reset` (ran on clean up) won't clean up the
+# CNI configuration and we must ensure a fresh environment before starting
+# Kubernetes.
+cleanup_cni_configuration() {
+	# Remove existing CNI configurations:
+	local cni_config_dir="/etc/cni"
+	local cni_interface="cni0"
+	sudo rm -rf /var/lib/cni/networks/*
+	sudo rm -rf "${cni_config_dir}"/*
+	if ip a show "$cni_interface"; then
+		sudo ip link set dev "$cni_interface" down
+		sudo ip link del "$cni_interface"
+	fi
+}
 
-# store iptables if CI running on bare-metal
-BAREMETAL="${BAREMETAL:-false}"
-iptables_cache="${KATA_TESTS_DATADIR}/iptables_cache"
-if [ "${BAREMETAL}" == true ]; then
+# Configure the cluster network.
+#
+# Parameters:
+#	$1 - path to the network plugin configuration file (Optional).
+#	     Defaults to flannel.
+#
+configure_network() {
+	local network_plugin_config="${1:-}"
+
+	if [ -z "${network_plugin_config}" ]; then
+		# default network plugin should be flannel, and its config file is taken from k8s 1.12 documentation
+		local flannel_version="$(get_test_version "externals.flannel.version")"
+		local flannel_url="$(get_test_version "externals.flannel.kube-flannel_url")"
+		echo "Use flannel ${flannel_version}"
+		network_plugin_config="$flannel_url"
+	fi
+	echo "Use configuration file from ${network_plugin_config}"
+	kubectl apply -f "$network_plugin_config"
+}
+
+# Save the current iptables configuration.
+#
+# Global variables:
+# 	KATA_TESTS_DATADIR - directory where to save the configuration (mandatory).
+#
+save_iptables() {
+	local iptables_cache="${KATA_TESTS_DATADIR}/iptables_cache"
+	[ -n "${KATA_TESTS_DATADIR}" ] || \
```

Comment:
Because of `nounset`, this will not trigger. If you insist on having the custom error message, you would have to do `KATA_TESTS_DATADIR`.

Reply:
Ah, true. I will have to use `${KATA_TESTS_DATADIR:-}`. I prefer a custom error message than the nounset error message.